modules/info.py

```diff
@@ -22,57 +22,117 @@
 """
 from __future__ import annotations
 
+from typing import TypeVar
+
 import discord
 from discord.ext import commands
 
 import core
+from constants import GUILD_ID
 
 
-class InformationEmbed(discord.Embed):
-    """A subclass of discord.Embed.
+EntityT = TypeVar("EntityT", discord.Member, discord.User, discord.Role, discord.abc.GuildChannel, discord.Guild)
 
-    This class allows to automatically get information within the class instead of recreating the embed each time
 
-    :param author: The embed author. Expects discord.Member or discord.User
-    :param entity: The Member, User, Role, TextChannel, or Guild to get information about.
-    """
+class NoPermissions(commands.CommandError):
+    pass
 
-    def __init__(
-        self,
-        *,
-        author: discord.Member | discord.User,
-        entity: discord.Member | discord.User | discord.Role | discord.TextChannel | discord.Guild,
-    ) -> None:
-        super().__init__()
-        created_at: str = f"{discord.utils.format_dt(entity.created_at)} ({discord.utils.format_dt(entity.created_at, 'R')})"
-        if isinstance(entity, discord.Member) and entity.joined_at:
-            joined_at = f"\n\nJoined At: {discord.utils.format_dt(entity.joined_at)} ({discord.utils.format_dt(entity.joined_at, 'R')})"
-        else:
-            joined_at = ""
 
-        description = f"Name: {entity.name}\n\nID: {entity.id}\n\nCreated At: {created_at}{joined_at}"
-        if isinstance(entity, discord.Member):
-            self.set_thumbnail(url=entity.guild_avatar or entity.display_avatar or None)
-        elif isinstance(entity, discord.User):
-            self.set_thumbnail(url=entity.display_avatar or None)
+class Information(core.Cog):
+    """Information commands which allows you to get information about users, the guild, roles, and channels."""
+
+    def __init__(self, bot: core.Bot) -> None:
+        self.bot: core.Bot = bot
+
+    async def cog_command_error(self, ctx: core.Context, error: commands.CommandError) -> None:  # type: ignore # bad lib types.
+        error = getattr(error, "original", error)
+
+        if isinstance(error, NoPermissions):
+            await ctx.send("Sorry, you don't have permissions to view details on this object.")
+            return
+
+    def _embed_factory(self, entity: EntityT) -> discord.Embed:
+        embed = discord.Embed(title=f"Info on {entity.name}!", colour=discord.Colour.random())
```

Comment:
Why are we using a random colour?

Reply:
It uses a random colour by default but certain factory methods do change it.
You can see that in `_member_info` and `_role_info` we take their colour if present.

If you'd prefer a more static default colour I'm happy to oblige.